[
    {
        "content": "<p>bvdh opened <a href=\"https://github.com/HL7/fhircast-docs/issues/341\">Issue #341</a>:</p>\n<blockquote>\n<p>When an application cannot follow a context change event, it will respond with a sync-error. This will notify the other applications that they are no longer in sync. <br>\nWhen the application has resolved the issue - how does it signal that context can be changed? Does it wait for a resend of the -open/-close event or can it actively signal this?</p>\n</blockquote>",
        "id": 216082270,
        "sender_full_name": "Github Notifications (FHIRcast)",
        "timestamp": 1604928011
    },
    {
        "content": "<p>isaacvetter <a href=\"https://github.com/HL7/fhircast-docs/issues/341#issuecomment-725685269\">commented</a> on <a href=\"https://github.com/HL7/fhircast-docs/issues/341\">Issue #341</a>:</p>\n<blockquote>\n<p>There is no way to currently actively signal the ending of a synchronization failure. It's a good idea. </p>\n<p>What would you name such an event? I suppose it would be broadcasted to systems that subscribed to it?</p>\n</blockquote>",
        "id": 216403998,
        "sender_full_name": "Github Notifications (FHIRcast)",
        "timestamp": 1605132063
    },
    {
        "content": "<p>bvdh <a href=\"https://github.com/HL7/fhircast-docs/issues/341#issuecomment-733713471\">commented</a> on <a href=\"https://github.com/HL7/fhircast-docs/issues/341\">Issue #341</a>:</p>\n<blockquote>\n<p>Some thoughts.</p>\n<p>A sync-error means that one or or more applications are not in sync - did not follow the event indicated by the event-id.<br>\nRe-establishing sync, means that all applications are in sync again. </p>\n<p>Root causes of sync-errors are:<br>\n1 application refuses change<br>\n2 an error in the application makes it impossible to process the change.<br>\n3 network error when sending the event<br>\n4 application has closed and did not unsubscribe</p>\n<p>The only mechanism in the current specification is that the sender of the original spec, will resend the event after a short time.</p>\n<p>Sync can only be re-established when the \"network\" is sound again - all application can receive events and are present. Topic 3 and 4 will be automatically handled as websockets will be closed and the application needs to resubscribe.</p>\n<p>2 should be handled by the hub - if an error occurs regularly, the hub should unsubscribe it. Which will happen anyway after the timeout.</p>\n<p>That leaves 1. The application could \"withdraw\" a syncerror with a syncrestored message. This could be a trigger to the hub to resend the original event (only when it is still the current one).<br>\nThe advantage of this approach is that no .write authorization is required for the application that sends the syncrestored message. Note that the resending of the Patient-open by original and hub are still valid options also in this situation.</p>\n<p>Conclusion:</p>\n<ul>\n<li>hubs SHALL resend an event that causes a sync-error, unless issue has been resolved by newer events.</li>\n<li>applications SHOULD resent an event that causes a sync-error</li>\n<li>applications SHOULD send a syncrestored event when the sync-error cause has been restored. This will trigger the hub to resent the original event (when relevant).</li>\n</ul>\n<p>Would this work&gt;</p>\n</blockquote>",
        "id": 217885207,
        "sender_full_name": "Github Notifications (FHIRcast)",
        "timestamp": 1606311721
    },
    {
        "content": "<p>gkustas <a href=\"https://github.com/HL7/fhircast-docs/issues/341#issuecomment-733835630\">commented</a> on <a href=\"https://github.com/HL7/fhircast-docs/issues/341\">Issue #341</a>:</p>\n<blockquote>\n<p>I just left a comment in #311 regarding syncerror. My scenario falls into your second root causes @bvdh - an error in the application (PowerScribe) that made it impossible to process the change. How would your proposal work in this situation? </p>\n<p>Like I said in the other thread, the end user sees the original error, and hopefully can rectify the error. In our case, the user will choose the appropriate study (accession from the proper RIS site) from the PowerScribe worklist, re-open the report causing another DiagnosticReport-open, and everyone would be back in sync again.</p>\n<p>Or do you see other scenarios where the syncerror is used effectively?</p>\n</blockquote>",
        "id": 217912889,
        "sender_full_name": "Github Notifications (FHIRcast)",
        "timestamp": 1606324172
    },
    {
        "content": "<p>bvdh <a href=\"https://github.com/HL7/fhircast-docs/issues/341#issuecomment-734143931\">commented</a> on <a href=\"https://github.com/HL7/fhircast-docs/issues/341\">Issue #341</a>:</p>\n<blockquote>\n<p>Sync-error will also be effective in indicating any of the other scenario's and can be used as a trigger to provide visual feedback to the user that sync is lost.<br>\nI think your scenario is good example of where an application due to a reason outside the scope of networking etc. refuses a context-change. I think that my solution will cover this. <br>\nAre we in sufficient agreement to move this into the spec (STU3 or STU2?).</p>\n</blockquote>",
        "id": 217974292,
        "sender_full_name": "Github Notifications (FHIRcast)",
        "timestamp": 1606378349
    }
]